translator/trace/zipkin/status_code.go

```diff
@@ -97,11 +98,25 @@ func (m *statusMapper) fromAttribute(key string, attrib *tracepb.AttributeValue)
 	return false
 }
 
-// attribToStatusCode maps an integer attribute value to a status code (int32).
-// The function return nil if the value is not an integer or an integer larger than what
-// can fit in an int32
-func attribToStatusCode(v *tracepb.AttributeValue) *int32 {
-	i := v.GetIntValue()
+// attribToStatusCode maps an integer or string attribute value to a status code.
+// The function return nil if the value is of another type or cannot be converted to an int32 value.
+func attribToStatusCode(attr *tracepb.AttributeValue) *int32 {
+	if attr == nil {
+		return nil
+	}
+
+	switch val := attr.Value.(type) {
+	case *tracepb.AttributeValue_IntValue:
+		return toInt32(int(val.IntValue))
+	case *tracepb.AttributeValue_StringValue:
+		if i, err := strconv.Atoi(val.StringValue.GetValue()); err == nil {
+			return toInt32(i)
+		}
+	}
+	return nil
+}
+
+func toInt32(i int) *int32 {
 	if i <= math.MaxInt32 {
```

Comment:
Should this also check `i >= math.MinInt32`?

Reply:
makes sense. added this check as well